### DISCUSSION TITLE
Can't get react-pdf to work with Astrojs

### DISCUSSION THREAD

**Discussion by tedpalmer1** [diegomura/react-pdf#1823] [Q&A] [answered]
Hi, does anyone have an example of using react-pdf with [Astro](https://astro.build/) client-side? I have been struggling and could really use some help. I am trying to use the package like so:

// src/components/samplePDF.jsx
```
import { Document, Page, PDFViewer, Text } from '@react-pdf/renderer'


const MyDocument = () => {

    return (
      <Document>
        <Page size='A4'>
            <Text>Test</Text>
        </Page>
      </Document>
    )
  }

const Preview = () => {

  return (
    <PDFViewer>
      <MyDocument/>
    </PDFViewer>
  )
}

export default Preview
```

// src/pages/sample-pdf/index.astro

```
---
import BaseLayout from '../../layouts/BaseLayout.astro'
import Nav from '../../components/Nav.jsx'
import Preview from '../../components/samplePDF.jsx'
---

<BaseLayout>
  <Nav client:load/>
  <Preview client:only/>
</BaseLayout>
```

The PDF is not being rendered and I am getting no errors

**Comment by jaroslaw-weber** [comment]
still not working in 2025 :(

**Comment by Enejivk** [comment]
I am having thesame problem with react pdf not showing in astro application

**Comment by Felipe-Navas** [comment, marked as answer]
Hi @Enejivk! 👋

You can check out the following repository, where I’ve added an example of how react-pdf is implemented and working as expected in an Astro project. Hope it helps!.

🔗 https://github.com/Felipe-Navas/test-react-pdf-on-astro
